src/inference/src/dev/core_impl.cpp

```diff
@@ -302,13 +332,17 @@ ov::Parsed ov::parseDeviceNameIntoConfig(const std::string& deviceName,
         }
     };
 
+    parsed._core_config.set(updated_config);
     // keep batch property only when called from query_supported_property
     if (!keep_core_property) {
-        clean_batch_properties(updated_device_name, updated_config, ov::hint::allow_auto_batching);
-        clean_batch_properties(updated_device_name, updated_config, ov::auto_batch_timeout);
+        for (const auto& name : auto_batch_properties_names) {
+            clean_batch_properties(updated_device_name, updated_config, name);
+        }
+        for (const auto& name : {ov::enable_mmap.name(), ov::force_tbb_terminate.name()}) {
```

Comment:
looks like it's safe to drop all properties from `core_properties_names` list here?

Reply:
BTW, if we have:
```c++
core.compile_model(filepath, "AUTO:CPU", enable_mmap(false));
```
then `enable_mmap` will be removed here ? and CPU will read model w/ mmap, right?

Reply:
Changed that core properties are removed for HW devices only, kept for others.
Add comment where `cache_dir` will be consumed from compile properties.